Improvements to crystal plasticity (TM) initialization

GregVernon opened the pull request:
Made some improvements to the TensorMechanics UserObject that reads in Euler angles and grain locations.  

References #11936 and #11939
<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

GregVernon committed:
Add two methods for assigning Euler angles to blocks

Closes #11936

GregVernon committed:
Add functionality for UserSpecified grain centers and Euler Angles. Refs #11936 #11939

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/217124/) on f3eca74 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11942/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11942/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format af8ec8e6cf0e693710321ab79c1e71a454fff4be`

GregVernon commented:
Not ready for primetime yet.

@jiangwen84 and I found and fixed a bug.  While we've fixed the bugs and it now works as intended (still need to push some commits), but we'll need to clean up the logic structure in ```ElementPropertyReadFile```.

GregVernon committed:
Set _npoints equal to # lines in grain center and euler angle files

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/217201/) on a248393 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11942/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11942/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format af8ec8e6cf0e693710321ab79c1e71a454fff4be`

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/218288/) on 52e1789 wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/11942/site)

This comment will be updated on new commits.

GregVernon committed:
Use MOOSE's built-in delimited file reader

This is in better alignment with MOOSE's principles

GregVernon committed:
initGrainCenterPoints should read points file

GregVernon committed:
Remove unused "_npoints" 

no longer needed, use "_ngrains" instead

GregVernon committed:
Rip out _npoints

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/218739/) on eb816c2 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](http://mooseframework.inl.gov/docs/PRs/11942/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s http://mooseframework.inl.gov/docs/PRs/11942/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format af8ec8e6cf0e693710321ab79c1e71a454fff4be`

GregVernon committed:
Fix typo
